azure_img_utils/cli/blob.py

```diff
@@ -0,0 +1,289 @@
+# -*- coding: utf-8 -*-
+
+"""Azure image utils cli module."""
+
+# Copyright (c) 2022 SUSE LLC
+#
+# This file is part of azure_img_utils. azure_img_utils provides an
+# api and command line utilities for handling images in the Azure Cloud.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import click
+import logging
+import sys
+
+from azure_img_utils.cli.cli_utils import (
+    add_options,
+    get_config,
+    process_shared_options,
+    shared_options,
+    echo_style
+)
+from azure_img_utils.azure_image import AzureImage
+
+
+# -----------------------------------------------------------------------------
+# Image commands function
+@click.group()
+def blob():
+    """
+    Commands for blob management.
+    """
+
+
+# -----------------------------------------------------------------------------
+# blob exists commands function
+@blob.command()
+@click.option(
+    '--storage-account',
+    type=click.STRING,
+    required=True,
+    help='Storage account for the blobs.'
+)
+@click.option(
+    '--blob-name',
+    type=click.STRING,
+    required=True,
+    help='Name of the blob to check.'
+)
+@click.option(
+    '--container',
+    type=click.STRING,
+    required=True,
+    help='Container for the blob to check.'
+)
+@add_options(shared_options)
+@click.pass_context
+def exists(
+    context,
+    storage_account,
+    blob_name,
+    container,
+    **kwargs
+):
+    """
+    Checks if a blob exists for the specified container
+    """
+
+    process_shared_options(context.obj, kwargs)
+    config_data = get_config(context.obj)
+    logger = logging.getLogger('azure_img_utils')
+    logger.setLevel(config_data.log_level)
+    try:
+        az_img = AzureImage(
+            container=container,
+            storage_account=storage_account,
```

Comment:
Since container and storage account are instance variables on the AzureImage class I would make them both shared optional args in the CLI similar to the resource group and the credentials_file. This way the user can set them in CLI or in config. This prevents duplication of logic because the class will handle validation if they are missing.

Reply:
Ditto for all the other commands

Reply:
Done!